manifest.go

```diff
@@ -61,46 +67,83 @@ func validateManifest(s string) ([]error, error) {
 		case "metadata":
 			// Check if metadata is of Map type
 			if reflect.TypeOf(val).Kind() != reflect.Map {
-				errs = append(errs, errors.New("metadata should be a TOML table"))
+				warns = append(warns, errors.New("metadata should be a TOML table"))
 			}
 		case "constraint", "override":
+			valid := true
 			// Invalid if type assertion fails. Not a TOML array of tables.
 			if rawProj, ok := val.([]interface{}); ok {
-				// Iterate through each array of tables
-				for _, v := range rawProj {
-					// Check the individual field's key to be valid
-					for key, value := range v.(map[string]interface{}) {
-						// Check if the key is valid
-						switch key {
-						case "name", "branch", "version", "source":
-							// valid key
-						case "revision":
-							if valueStr, ok := value.(string); ok {
-								if abbrevRevHash.MatchString(valueStr) {
-									errs = append(errs, fmt.Errorf("revision %q should not be in abbreviated form", valueStr))
+				// Check element type. Must be a map. Checking one element would be
+				// enough because TOML doesn't allow mixing of types.
+				if reflect.TypeOf(rawProj[0]).Kind() != reflect.Map {
```

Comment:
I checked with just a single empty `[[constraint]]` in manifest and no constraint element.

When no constraint, `"constraint"` case (switch-case) is never executed.
When there are one or more constraint, `"constraint"` case is executed and there's at least one element in the array.

TOML table adds an empty map, no matter if the table has any content or empty. Safe enough.